AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/script_canvas_tools_qt.py

```diff
@@ -0,0 +1,448 @@
+    action.trigger()
+
+def save_script_event_file(self, file_path):
+    """
+    function for saving a script event file with a user defined file path. Requires asset editor qt object to be initialized
+    and any required fields in the asset editor to be filled in before asset can be saved.
+
+    param self: the script object calling this function
+    param file_path: full path to the file as a string
+
+    returns: true if the Save action is successful and the * character disappears from the asset editor label
+    """
+    editor.AssetEditorWidgetRequestsBus(bus.Broadcast, SAVE_ASSET_AS, file_path)
+    action = pyside_utils.find_child_by_pattern(self.asset_editor_menu_bar, {"type": QtWidgets.QAction, "iconText": SAVE_STRING})
+    action.trigger()
+    # wait till file is saved, to validate that check the text of QLabel at the bottom of the AssetEditor,
+    # if there are no unsaved changes we will not have any * in the text
+    label = self.asset_editor.findChild(QtWidgets.QLabel, "textEdit")
+    return helper.wait_for_condition(lambda: "*" not in label.text(), WAIT_TIME_3)
+
+
+def initialize_asset_editor_object(self):
+    """
+    function for initializing qt objects needed for testing around asset editor
+
+    param self: the script object calling this function.
+
+    returns: None
+    """
+    self.asset_editor = self.editor_main_window.findChild(QtWidgets.QDockWidget, ASSET_EDITOR_UI)
+    self.asset_editor_widget = self.asset_editor.findChild(QtWidgets.QWidget, "AssetEditorWindowClass")
+    self.asset_editor_row_container = self.asset_editor_widget.findChild(QtWidgets.QWidget, "ContainerForRows")
+    self.asset_editor_menu_bar = self.asset_editor_widget.findChild(QtWidgets.QMenuBar)
+
+
+def initialize_node_palette_object(self):
+    """
+    function for initializing qt objects needed for testing around the script canvas editor
+
+    param self: the script object calling this function
+
+    returns: None
+    """
+    self.node_palette = self.sc_editor.findChild(QtWidgets.QDockWidget, NODE_PALETTE_QT)
+    self.node_tree_view = self.node_palette.findChild(QtWidgets.QTreeView, TREE_VIEW_QT)
+    self.node_tree_search_frame = self.node_palette.findChild(QtWidgets.QFrame, SEARCH_FRAME_QT)
+    self.node_tree_search_box = self.node_tree_search_frame.findChild(QtWidgets.QLineEdit, SEARCH_FILTER_QT)
+
+
+def expand_qt_container_rows(self, object_name):
+    """
+    function used for expanding qt container rows with expandable children
+
+    param self: The script object calling this function
+    param object_name: qt object name as a string
+
+    returns: none
+    """
+    children = self.asset_editor_row_container.findChildren(QtWidgets.QFrame, object_name)
+    for child in children:
+        check_box = child.findChild(QtWidgets.QCheckBox)
+        if check_box and not check_box.isChecked():
+            QtTest.QTest.mouseClick(check_box, QtCore.Qt.LeftButton, QtCore.Qt.NoModifier)
+
+
+def open_node_palette(self):
```

Comment:
Can this be refactorted into SC Editor or its own QtPy class?

Reply:
Yep! I plan on putting all the sc tools into their own classes and adding references to them on the SC editor.